core/time/timezone/tz_unix.odin

```diff
@@ -0,0 +1,91 @@
+#+build darwin, linux, freebsd, openbsd, netbsd
+package timezone
+
+import "core:os"
+import "core:strings"
+import "core:path/filepath"
+import "core:time/datetime"
+
+local_tz_name :: proc(allocator := context.allocator) -> (name: string, success: bool) {
+	local_str, ok := os.lookup_env("TZ", allocator)
+	if !ok {
+		orig_localtime_path := "/etc/localtime"
+		path, err := os.absolute_path_from_relative(orig_localtime_path)
+		if err != nil {
+			// If we can't find /etc/localtime, fallback to UTC
+			if err == .ENOENT {
+				return strings.clone("UTC", allocator), true
+			}
+
+			return
+		}
+		defer delete(path)
+
+		// FreeBSD makes me sad.
+		if path == orig_localtime_path {
+			data, ok2 := os.read_entire_file_from_filename("/var/db/zoneinfo")
+			if !ok2 {
+				return
+			}
+			defer delete(data)
+
+			tzone_name := strings.clone(strings.trim_space(string(data)), allocator)
+			return tzone_name, true
+		}
+
+		// Looking for tz path (ex fmt: "UTC", "Etc/UTC" or "America/Los_Angeles")
+		path_dir, path_file := filepath.split(path)
+		if path_dir == "" {
+			return
+		}
+		upper_path_dir, upper_path_chunk := filepath.split(path_dir[:len(path_dir)-1])
+		if upper_path_dir == "" {
+			return
+		}
+
+		tzone_name : string
+		if strings.contains(upper_path_chunk, "zoneinfo") {
+			tzone_name = strings.clone(path_file, allocator)
+		} else {
+			tzone_name = filepath.join([]string{upper_path_chunk, path_file})
+		}
+
+		return tzone_name, true
+	}
+
+	if local_str == "" {
+		return strings.clone("UTC", allocator), true
+	}
+
+	return local_str, true
+}
+
+_region_load :: proc(reg_str: string, allocator := context.allocator) -> (out_reg: ^datetime.TZ_Region, success: bool) {
```

Comment:
Same comment as the proc above where you may want to split main allocation and temporary allocation.